### PR TITLE
Respect limit field passed in ListRequests

### PR DESCRIPTION
**Pull request by andrewnester**
## Changes
Respect limit field passed in ListRequests

## Tests
Ran queries to Jobs with limit options set to none, 0, 5, 20, 25 and 50.

- [x] `make test` passing
- [x] `make fmt` applied
- [x] relevant integration tests applied

```
--- PASS: TestAccJobsListWithLimit (9.64s)
PASS
ok      github.com/databricks/databricks-sdk-go/internal        10.481s
```

### CODE REVIEW

**Comment by pietern** [comment]
@andrewnester Please also include an integration test where you assert on the size of the returned slice.

**Comment by andrewnester** [comment]
@pietern added integration test

File: internal/jobs_test.go [hunk 1]
```diff
@@ -191,3 +191,60 @@ func TestAccJobsListAllNoDuplicatesNoTranspile(t *testing.T) {
 	}
 	assert.Equal(t, len(all), len(ids), "Listing produced duplicate results")
 }
+
+func TestAccJobsListWithLimit(t *testing.T) {
```
Comment (nfx) [inline internal/jobs_test.go:195]:
```suggestion
func TestAccJobsListWithLimitNoTranspile(t *testing.T) {
```

to properly work with test cross-compilation infra, as we don't yet support transpiling loops

File: .codegen/api.go.tmpl [hunk 2]
```diff
@@ -144,6 +151,12 @@ func (a *{{.Service.Name}}API) {{.PascalName}}All(ctx context.Context{{if .Reque
 		{{- else -}}
 		request.{{.Pagination.Offset.PascalName}} += {{template "type" .Pagination.Offset.Entity}}(len(response.{{.Pagination.Results.PascalName}}))
 		{{- end}}
+		{{- if .Pagination.Limit }}
```
Comment (nfx) [inline .codegen/api.go.tmpl:154]:
```suggestion
		{{- if .Pagination.Limit -}}
```

File: .codegen/api.go.tmpl [hunk 3]
```diff
@@ -104,6 +104,9 @@ func (a *{{.Service.Name}}API) {{.PascalName}}AndWait(ctx context.Context{{if .R
 // This method is generated by Databricks SDK Code Generator.
 func (a *{{.Service.Name}}API) {{.PascalName}}All(ctx context.Context{{if .Request}}, request {{.Request.PascalName}}{{end}}) ([]{{.Pagination.Entity.PascalName}}, error) {
 	{{if .Pagination.MultiRequest}}var results []{{.Pagination.Entity.PascalName}}
+	{{ if .Pagination.Limit }}
```
Comment (nfx) [inline .codegen/api.go.tmpl:107]:
```suggestion
	{{ if .Pagination.Limit -}}
```

File: .codegen/api.go.tmpl [hunk 4]
```diff
@@ -129,6 +132,10 @@ func (a *{{.Service.Name}}API) {{.PascalName}}All(ctx context.Context{{if .Reque
 			{{- end}}
 			results = append(results, v)
 		}
+		{{ if .Pagination.Offset }}
```
Comment (nfx) [inline .codegen/api.go.tmpl:135]:
```suggestion
		{{ if .Pagination.Offset -}}
```

**Review by nfx — CHANGES_REQUESTED** [review]
Whitespace needs to be a bit more consistent

**Review by nfx — APPROVED** [review]

**Pull request merged**